src/test/java/bdd/steps/NameMatchingSteps.java

```diff
@@ -555,4 +582,162 @@ private static ObjectMapper setupObjectMapper() {
+                       n.diacritics() == inputName.containsDiacritics() &&
+                       n.umlauts() == inputName.containsUmlauts() &&
+                       n.abbreviation() == inputName.abbreviation() &&
+                       n.nameSplitter() == inputName.containsNameSplitter() &&
+                       n.uniCodeBlocks().containsAll(inputName.unicodeBlocks()) &&
+                       inputName.unicodeBlocks().containsAll(n.uniCodeBlocks()) &&
+                       n.length() == inputName.getLength();
+    }
+
+    @And("^the meta-data indicates that the following generators were used$")
+    public void theMetaDataIndicatesThatTheFollowingGeneratorsWereUsed(DataTable dataTable) {
+
+        List<String> rawGenerators = dataTable.asList(String.class);
+        List<Generator> generators = asGenerators(rawGenerators);
+
+        verify(mockAppender).doAppend(argThat(argument -> {
+            LoggingEvent loggingEvent = (LoggingEvent) argument;
+
+            return matchAchieved(loggingEvent) &&
+                           metaDataWasLogged(loggingEvent) &&
+                           (metaDataHasExpectedNumberOfGenerators(generators, loggingEvent) || diagnoseWrongGenerator(generators, loggingEvent)) &&
+                           (metaDataHasGenerators(generators, loggingEvent) || diagnoseWrongGenerator(generators, loggingEvent));
+        }));
+    }
+
+    private List<Generator> asGenerators(List<String> rawGenerators) {
+        return rawGenerators.stream()
+                       .map(s -> asList(s.split("\\s+")))
+                       .flatMap(Collection::stream)
+                       .map(s -> Generator.valueOf(s))
+                       .collect(toList());
+    }
+
+    private boolean diagnoseWrongGenerator(List<Generator> expected, LoggingEvent loggingEvent) {
+        CandidateDerivation candidateDerivation = getCandidateDerivation(loggingEvent);
+        log.error("Expected: {} Actual: {}", expected, candidateDerivation.generators());
+        return false;
+    }
+
+    private boolean metaDataHasExpectedNumberOfGenerators(List<Generator> generators, LoggingEvent loggingEvent) {
+        CandidateDerivation candidateDerivation = getCandidateDerivation(loggingEvent);
+        return generators.size() == candidateDerivation.generators().size();
+    }
+
+    private boolean metaDataHasGenerators(List<Generator> generators, LoggingEvent loggingEvent) {
+        CandidateDerivation candidateDerivation = getCandidateDerivation(loggingEvent);
+        return generators.equals(candidateDerivation.generators());
+    }
+
+    private CandidateDerivation getCandidateDerivation(LoggingEvent loggingEvent) {
+        return (CandidateDerivation) ReflectionTestUtils.getField(loggingEvent.getArgumentArray()[2], "object");
+    }
+
+    @And("^the meta-data contains the following name derivation information$")
+    public void theMetaDataContainsTheFollowingNameDerivationInformation(DataTable dataTable) {
```

Comment:
The diagnosis done on the input names and generators makes test failures easier to understand - can we do the same with the name derivations?

Reply:
Agreed